plugins/plugin_utils/terminalmanager.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: Contributors to the Ansible project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+import random
+import re
+import string
+
+from ansible.module_utils._text import to_bytes
+from ansible.module_utils._text import to_text
+from ansible.plugins.shell.powershell import _common_args
+
+
+class TerminalManager:
+    def __init__(self, connection):
+        self.connection = connection
+
+    def prepare_terminal(self) -> None:
+        """perform any one-time terminal settings"""
+        # No Windows setup for now
+        if self.connection.is_windows:
+            return
+
+        # Ensure SSM Session has started
+        self.ensure_ssm_session_has_started()
+
+        # Disable echo command
+        self.disable_echo_command()  # pylint: disable=unreachable
+
+        # Disable prompt command
+        self.disable_prompt_command()  # pylint: disable=unreachable
+
+        self.connection.verbosity_display(4, "PRE Terminal configured")  # pylint: disable=unreachable
+
+    def wrap_command(self, cmd: str, mark_start: str, mark_end: str) -> str:
+        """Wrap command so stdout and status can be extracted"""
+
+        if self.connection.is_windows:
+            if not cmd.startswith(" ".join(_common_args) + " -EncodedCommand"):
+                cmd = self.connection._shell._encode_script(cmd, preserve_rc=True)
+            cmd = cmd + "; echo " + mark_start + "\necho " + mark_end + "\n"
+        else:
+            cmd = (
+                f"printf '%s\\n' '{mark_start}';\n"
+                f"echo | {cmd};\n"
+                f"printf '\\n%s\\n%s\\n' \"$?\" '{mark_end}';\n"
+            )  # fmt: skip
+
+        self.connection.verbosity_display(4, f"wrap_command: \n'{to_text(cmd)}'")
+        return cmd
+
+    def disable_echo_command(self) -> None:
+        """Disable echo command from the host"""
+        disable_echo_cmd = to_bytes("stty -echo\n", errors="surrogate_or_strict")
```

Comment:
If stty -echo fails, the method might hang indefinitely. What do you think about introducing a timeout or a maximum number of retries?

Reply:
There is already a timeout: https://github.com/ansible-collections/community.aws/blob/2778c9fcec3762c3765657e85e38130f44d726ea/plugins/connection/aws_ssm.py#L666-L674